src/arch/x86/cpuid.cc

```diff
@@ -31,162 +31,146 @@
 #include "arch/x86/isa.hh"
 #include "base/bitfield.hh"
 #include "cpu/thread_context.hh"
+#include "debug/X86.hh"
 
 namespace gem5
 {
 
-namespace X86ISA {
-    enum StandardCpuidFunction
-    {
-        VendorAndLargestStdFunc,
-        FamilyModelStepping,
-        CacheAndTLB,
-        SerialNumber,
-        CacheParams,
-        MonitorMwait,
-        ThermalPowerMgmt,
-        ExtendedFeatures,
-        NumStandardCpuidFuncs
-    };
-
-    enum ExtendedCpuidFunctions
-    {
-        VendorAndLargestExtFunc,
-        FamilyModelSteppingBrandFeatures,
-        NameString1,
-        NameString2,
-        NameString3,
-        L1CacheAndTLB,
-        L2L3CacheAndL2TLB,
-        APMInfo,
-        LongModeAddressSize,
-
-        /*
```

Comment:
seems like these are still not implemented and should have been left?

Reply:
The reason I decided to remove these is because there are dozens of missing functions so I don't think these ones are in any way special.  In a patch in this "rel change" (if we can still call it that), it sets an enum to a value which makes the compiler skip the other values. I am guessing the original author may have added these so that the final/last enum value returns the number of functions to the CPUID call correctly

Reply:
Ok ... I don't know this code well enough to say if this is a good idea or not, but I guess at least GitHub will document it here for future reference.

Reply:
There are dozens of CPUID functions missing and I see no reason to define them here.  Since it's possible to just assign a number to the enum name, I suggest using that rather than having these placeholder functions that aren't actually implemented just to get the number to line up.